### ISSUE TITLE
Why do untyped private constants have such narrow type?

### ISSUE THREAD

**Issue by dimaqq** [microsoft/pyright#9828]
```py
🦐~> uvx pyright Downloads/repro.py
Installed 3 packages in 142ms
/Users/dima/Downloads/repro.py
  /Users/dima/Downloads/repro.py:6:5 - error: Cannot assign to attribute "_CONST" for class "Foo"
    Expression of type "Literal[13]" cannot be assigned to attribute "_CONST" of class "Foo"
      "Literal[13]" is not assignable to type "Literal[42]" (reportAttributeAccessIssue)
1 error, 0 warnings, 0 informations


🦐~ [1]> cat Downloads/repro.py
class Foo:
    _CONST = 42


foo = Foo()
foo._CONST = 13
```

**Comment by dimaqq** [comment]
pyright 1.1.393

**Comment by erictraut** [comment]
Constants cannot be overwritten (by definition), so they should have the narrowest (most accurate) type possible.

If you want to reassign the value, you can use a lowercase name, in which case pyright will treat it as a variable.
